Userland/grep.cpp

```diff
@@ -47,75 +62,108 @@ int main(int argc, char** argv)
     bool recursive { false };
     bool use_ere { true };
     const char* pattern = nullptr;
+    const char* binary_mode_str = "binary";
+    bool case_insensitive = false;
 
     Core::ArgsParser args_parser;
     args_parser.add_option(recursive, "Recursively scan files starting in working directory", "recursive", 'r');
     args_parser.add_option(use_ere, "Extended regular expressions (default)", "extended-regexp", 'E');
     args_parser.add_option(pattern, "Pattern", "regexp", 'e', "Pattern");
+    args_parser.add_option(case_insensitive, "Make matches case-insensitive", nullptr, 'i');
+    args_parser.add_option(binary_mode_str, "Action to take for binary files ([binary], text, skip)", "binary-mode", 0, "mode");
+    args_parser.add_option(Core::ArgsParser::Option {
+        .requires_argument = false,
+        .help_string = "Treat binary files as text (same as --binary-mode text)",
+        .long_name = "text",
+        .short_name = 'a',
+        .accept_value = [&](auto) {
+            binary_mode_str = "text";
+            return true;
+        },
+    });
+    args_parser.add_option(Core::ArgsParser::Option {
+        .requires_argument = false,
+        .help_string = "Ignore binary files (same as --binary-mode skip)",
+        .long_name = nullptr,
+        .short_name = 'I',
+        .accept_value = [&](auto) {
+            binary_mode_str = "skip";
+            return true;
+        },
+    });
```

Comment:
Cool that you've mastered advanced args parser :smile:

Since you're doing that, perhaps get rid of `binary_mode_str` and just save to `binary_mode` directly everywhere? If you get a string you don't recognize, `return false` to args parser and it'll arrange to display an error and die.

Reply:
That's a better idea, yeah!